src/ripple/app/ledger/impl/InboundLedger.cpp

```diff
@@ -160,25 +167,12 @@ InboundLedger::init(ScopedLockType& collectionLock)
 std::size_t
 InboundLedger::getPeerCount() const
 {
-    return std::count_if(mPeers.begin(), mPeers.end(), [this](auto id) {
-        return app_.overlay().findPeerByShortID(id) != nullptr;
+    std::size_t count = 0;
+    mPeerSet->visitAddedPeers([this, &count](auto id) {
+        if (app_.overlay().findPeerByShortID(id) != nullptr)
+            ++count;
```

Comment:
We could go back to using standard algorithms if `PeerSet` was a container, which I think it should be. Short of that, it could just expose a `const` view of the privately-held container.

`PeerSet` is never shared, especially now that it is composed instead of derived. Only the surrounding types are shared, and they already have a lock for their internals that could be used to guard `PeerSet` just like it guards any other member. Then we could get rid of `PeerSet`'s lock.

Reply:
Thanks, now the privately-held container is exposed as a `const&`. 

The lock is removed as discussed.